src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py

```diff
@@ -239,17 +241,21 @@ def register_source_uri_arguments(scope):
 for item in ['blob', 'file', 'queue', 'table']:
     register_cli_argument('storage account show-connection-string', '{}_endpoint'.format(item), help='Custom endpoint for {}s.'.format(item))
 
+register_cli_argument('storage account create', 'account_type', help='The storage account type', **enum_choice_list(get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='AccountType')))
+
 register_cli_argument('storage account create', 'account_name', account_name_type, options_list=('--name', '-n'), completer=None)
 
-register_cli_argument('storage account create', 'kind', help='Indicates the type of storage account.', **enum_choice_list(Kind))
+register_cli_argument('storage account create', 'kind', help='Indicates the type of storage account.', **enum_choice_list(get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='Kind')))
```

Comment:
I second what @tjprescott said. The pattern is way to confusing and unintuitive.